src/modifier/init-modifier-types.ts

```diff
@@ -211,10 +211,13 @@ export function initModifierTypes() {
 
   modifierTypes.BERRY = () =>
     new ModifierTypeGenerator((_party: Pokemon[], pregenArgs?: any[]) => {
-      if (pregenArgs && pregenArgs.length === 1 && pregenArgs[0] in BerryType) {
+      if (pregenArgs && pregenArgs.length === 1 && Object.values(BerryType).includes(pregenArgs[0])) {
         return new BerryModifierType(pregenArgs[0] as BerryType);
       }
-      const berryTypes = getTSEnumValues(BerryType);
+      const berryTypes = Object.values(BerryType).filter((v) => {
+        const removeBerries: BerryType[] = [BerryType.SITRUS, BerryType.LUM, BerryType.LEPPA] as const;
```

Comment:
```suggestion
      const removeBerries: BerryType[] = [BerryType.SITRUS, BerryType.LUM, BerryType.LEPPA] as const;
      const berryTypes = Object.values(BerryType).filter((v) => {
```
just so that it doesn't recreate the array for every berry

Reply:
Ah, good point.